connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -613,7 +614,7 @@ public static ConfigInfos generateResult(String connType, Map<String, ConfigKey>
         return new ConfigInfos(connType, errorCount, groups, configInfoList);
     }
 
-    private static ConfigKeyInfo convertConfigKey(ConfigKey configKey) {
+    public static ConfigKeyInfo convertConfigKey(ConfigKey configKey) {
```

Comment:
Nit: kind of strange that this method (and honestly a lot of the config wrangling logic) is contained in the `AbstractHerder` class. Might be worth refactoring into a separate config utils class/package. Probably best to leave for a separate PR, but if you agree that that'd be cleaner I could see about doing that work as a follow-up.